src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -258,17 +259,18 @@ class MetadataBasedItemList extends React.Component<Props, State> {
                     return cellData;
                 }
                 const { type, value, options = [] } = field;
+                const shouldShowEditIcon = isCellEditable && isString(type);
                 cellData = (
                     <>
                         {!isCellBeingEdited && <ReadOnlyMetadataField dataValue={value} displayName="" type={type} />}
-                        {isCellEditable && (
+                        {shouldShowEditIcon && (
                             <IconWithTooltip
                                 type={EDIT_ICON_TYPE}
                                 tooltipText={<FormattedMessage {...messages.editLabel} />}
                                 onClick={() => this.handleEditIconClick(columnIndex, rowIndex, value)}
                             />
                         )}
-                        {isCellBeingEdited && valueBeingEdited && (
```

Comment:
Show edit icon even for the case where value is not present to start with; i.e. for `JSON patch add` operation

Reply:
Is there a snapshot test for this change?

Reply:
@sahiga there is but its difficult to test react-virtualized `<AutoSizer />` component with enzyme because of issues like [this](https://github.com/bvaughn/react-virtualized/issues/493#issuecomment-265020587). The other alternative I found out was to export one more component from the same file which is wrapped in `<VirtualScroll />` as mentioned [here](https://stackoverflow.com/questions/37801005/does-react-virtualized-work-with-airbnb-enzyme) so that we can test the functionality but I guess its too complicated for a snapshot test. I have instead relied on unit testing other functionality from the component.